colcon_core/shell/template/prefix.bat.em

```diff
@@ -53,6 +57,36 @@ goto:eof
 goto:eof
 
 
+:: Get the package names in topological order
+:: using semicolons as separators and avoiding leading separators.
+:: first argument: the name of the result variable
+:: second argument: the base path to look for packages
+:ament_get_ordered_packages
+  setlocal enabledelayedexpansion
+
+  :: use the Python executable known at configure time
+  set "_colcon_python_executable=@(python_executable)"
+  :: allow overriding it with a custom location
+  if "%COLCON_PYTHON_EXECUTABLE%" NEQ "" (
+    set "_colcon_python_executable=%COLCON_PYTHON_EXECUTABLE%"
+  )
+
+  set "_colcon_ordered_packages="
+  for /f %%p in ('""%_colcon_python_executable%" "%~dp0_local_setup_util.py"@
```

Comment:
there's a trailing @ here that might be an accident (or maybe it's just to cut off the newline even if there isn't any other empy stuff in this line): I just haven't seen it before so wanted to draw attention to it

Reply:
That is intentionally. In an `empy` template this will suppress the following character which in this case is the newline. The result is that the optional `--merge-install` appears on the same line as the Python invocation. The `for` loop line extends from line 75 to line 79 in the template but results in a single line in the output.

Reply:
OK that's what I was thinking after a while but it's good to get clarification thanks!